### PR TITLE
Component API for UDP driver

### PR DESCRIPTION
**Pull request by esteve**
Still a WIP

### CODE REVIEW

**Comment by JWhitleyWork** [comment]
@esteve Is this still a WIP?

**Comment by esteve** [comment]
@JWhitleyWork tests pass here, but not for downstream packages (i.e. Autoware.Auto), so I'm not confident to submit this for review yet.

File: udp_driver/include/udp_driver/udp_driver_node.hpp [hunk 1]
```diff
@@ -107,30 +109,48 @@ class UdpDriverNode : public rclcpp::Node
...
+      rclcpp::spin_some(this->get_node_base_interface());
+      if (m_timer->is_canceled()) {
+        return;
       }
-      ++iter;
-      try {
-        PacketT pkt;
-        (void) get_packet(pkt, m_udp_socket);
+    }
+  }
```
Comment (JWhitleyWork) [inline udp_driver/include/udp_driver/udp_driver_node.hpp:123]: So, I don't really know that this is the right way to go about this. A node spinning itself essentially means that the handling of callbacks and timers is out of the control of the parent thread. Especially in this case where this class is meant to be inherited, I don't think this is the pattern we want.

I think my recommendation would be to let the inheriting node handle the spinning and/or running the read function in-general. Drop the while loop and just expose the `run` function to the child class and let them figure out how often to run it and for how long.

Comment (esteve) [reply udp_driver/include/udp_driver/udp_driver_node.hpp:123]: Thanks for the feedback. This method is meant as a transition for downstream drivers, I didn't want to break the API, but with a `LoopingCall` I needed a way to trigger the executor to continue processing. I agree with you about letting inheriting nodes handle the spinning, I'd do it in a different method and mark the `run` method deprecated.

**Comment by esteve** [comment]
Withdrawing since the API has changed significantly since I created this PR.

**Pull request closed without merging**